Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
+            switch (notification.Data.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+                    await CreatePromoPoll(notification.Data);
+                    break;
+                case PromotionActionType.CampaignClosed:
+                    await DeletePromoPoll(notification.Data);
+                    break;
+            }
+        }
+
+        public async Task HandleNotificationAsync(ReactionAddedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
+
+            if (cancellationToken.IsCancellationRequested
+                || notification.Reaction.User.Value.IsBot
+                || notification.Message.Value.Embeds.Count < 1)
+                return;
+
+            if (_memoryCache.TryGetValue(notification.Message.Id, out CachedPromoPoll poll)
+            && ValidateVoteReaction(reaction))
+            {
+                await notification.Message.Value.RemoveReactionAsync(notification.Reaction.Emote, notification.Reaction.User.Value);
+
+                var sentiment = reaction.Name == _approveEmoji.Name ? PromotionSentiment.Approve : PromotionSentiment.Oppose;
+
+                if (!await _promotionsService.VotePromoAsync(poll.Campaign.Id, user,  sentiment))
+                    return;
+
+                if (reaction.Name == _approveEmoji.Name)
+                    ++poll.Approve;
+                else if(reaction.Name == _disproveEmoji.Name)
+                    ++poll.Disprove;
+
+                await poll.Message.ModifyAsync(m => m.Embed = BuildPollEmbed(poll.Campaign, poll.Approve, poll.Disprove).Build());
+            }
+        }
+
+        private async Task CreatePromoPoll(PromotionActionCreationData campaign)
+        {
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+            var message = await pollChannel.SendMessageAsync(embed: BuildPollEmbed(campaign.Campaign, 1, 0).Build());
+
+            await message.AddReactionsAsync(new[] {(IEmote)_approveEmoji, _disproveEmoji});
+
+            var poll = new CachedPromoPoll
+            {
+                Message = message,
+                Campaign = campaign.Campaign,
+                Approve = 1,
+                Disprove = 0
+            };
+
+            _memoryCache.Set(campaign.Campaign.Id, poll);
+            _memoryCache.Set(message.Id, poll);
+        }
+
+        private async Task DeletePromoPoll(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(campaign.Campaign.Id, out CachedPromoPoll poll))
+                return;
+            await poll.Message.DeleteAsync();
```

Comment:
Should we instead just update the message to say that the campaign has closed, and remove the buttons?

Reply:
this will expose the campaign vote counts to the person who was promoted. which could cause hurt feelings etc. we could just remove the vote counts on promo close but that seems redundant. We already have a log of the promotions. no need to have it in two places.

Reply:
I suppose that fits well enough with calling it the "Promotions Interface" channel.